plugins/memblaze/memblaze-nvme.c

```diff
@@ -481,3 +481,129 @@ static int set_additional_feature(int argc, char **argv, struct command *cmd, st
 		free(buf);
 	return err;
 }
+
+static int memblaze_fw_commit(int fd, int select)
+{
+	struct nvme_admin_cmd cmd = {
+		.opcode		= nvme_admin_activate_fw,
+		.cdw10		= 8,
+		.cdw12      = select,
+	};
+
+	return nvme_submit_admin_passthru(fd, &cmd);
+}
+
+static int memblaze_selective_download(int argc, char **argv, struct command *cmd, struct plugin *plugin)
+{
+	const char *desc = "This performs a selective firmware download, which allows the user to " \
+			"select which firmware binary to update. This requires a power cycle once the " \
+			"update completes. The options available are: \n\n" \
+			"OOB - This updates the OOB and main firmware\n" \
+			"EEP - This updates the eeprom and main firmware\n" \
+			"ALL - This updates the eeprom, OOB, and main firmware";
+	const char *fw = "firmware file (required)";
+	const char *select = "FW Select (e.g., --select=ALL)";
+	int xfer = 4096;
+	void *fw_buf;
+	int fd, selectNo,fw_fd,fw_size,err,offset = 0;
+	struct stat sb;
+	int i;
+
+	struct config {
+		char  *fw;
+		char  *select;
+	};
+
+	struct config cfg = {
+		.fw     = "",
+		.select = "\0",
+	};
+
+	OPT_ARGS(opts) = {
+		OPT_FILE("fw",     'f', &cfg.fw,     fw),
+		OPT_LIST("select", 's', &cfg.select, select),
```

Comment:
Why is this a list option? That's what you'd use if you want to have multiple comma separated values, like "--select=OOB,EEP".

Reply:
I updated prompt.
--select=OOB,EEP,ALL

Reply:
> I updated prompt.
> --select=OOB,EEP,ALL

I still don't understand. It doesn't look like you support a list at all. You're enforcing it be exactly 3 characters long rather than a comma separated value. Why not just make this an OPT_STRING instead?